src/data/periods.js

```diff
@@ -0,0 +1,26 @@
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Date Periods. This is here to accomodate the split build process. The typescript copy should supersede this when appropriate.
+ */
+const Periods = [
```

Comment:
Typically the capitalisation is reserved for react components and names of type definitions. I think it may have been confusing where you saw `const periods: Periods = [ .... ]` - `periods` is the variable name, and `Periods` is the type definition name.